vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt

```diff
@@ -1,14 +1,22 @@
+# Set variables.
 set(lpc54018_dir "${AFR_VENDORS_DIR}/nxp/LPC54018")
 set(lpc54018iotmodule_dir "${AFR_VENDORS_DIR}/nxp/boards/lpc54018iotmodule")
 
 set(lpc54018_ports_dir "${lpc54018iotmodule_dir}/ports")
-set(lpc54018_demos_common_dir "${lpc54018iotmodule_dir}/aws_demos/common")
-set(lpc54018_tests_common_dir "${lpc54018iotmodule_dir}/aws_tests/common")
 
 if(AFR_IS_TESTING)
-    set(lpc54018_app_dir "${lpc54018_tests_common_dir}")
+    set(lpc54018_aws_dir "${lpc54018iotmodule_dir}/aws_tests")
 else()
-    set(lpc54018_app_dir "${lpc54018_demos_common_dir}")
+    set(lpc54018_aws_dir "${lpc54018iotmodule_dir}/aws_demos")
+endif()
+
+# Include IDE specific cmake file.
+if(${AFR_TOOLCHAIN} STREQUAL "arm-gcc")
+    include("mcuxpresso.cmake")
+elseif(${AFR_TOOLCHAIN} STREQUAL "arm-iar") 
+    include("iar.cmake")
+else()
+    message( FATAL_ERROR "The toolchain is not supported." )
```

Comment:
Throwing a fatal error here would be an issue for OCW. They want to run it in configuration mode only which use the default host compiler (gcc for x86 on linux). I have a variable for OCW `AFR_NON_BUILD_MODE`, you can instead check,
```cmake
elseif(NOT AFR_NON_BUILD_MODE)
    message( FATAL_ERROR "The toolchain is not supported." )
endif()
```

Reply:
Got it. Let me update.